### DISCUSSION TITLE
Jamulus explorer not giving response

### DISCUSSION THREAD

**Discussion by Meemansa-Mishra** [jamulussoftware/jamulus#2959] [Technical Support & Help]
I have used "**https://explorer.jamulus.io/servers.php?directory=**" url to get a list of all hosted servers on my directory but have been having a lot of trouble recently because I don't receive response from the url everytime.
I also noticed same problems with the new urls:
**https://explorer.jamulus.io/servers-sf.php?directory=**
**https://explorer.jamulus.io/servers-lon2.php?directory=**
Any other service or api that I can use to get my server list?

**Comment by rdica** [comment]
> Any other service or api that I can use to get my server list?

Configure your custom directory in the jamulus client settings.

I spun up a test directory and had no issues viewing it with the above URLs.

Check your network or firewall, you may be blocking or otherwise limiting access to the three IPs in use by explorer.jamulus.io:
- 80.85.86.7 London Linode
- 18.133.158.130 London AWS
- 147.182.226.170 San Francisco

**Reply by Meemansa-Mishra** [reply]
@rdica 
The issue is the we don't always get a response back.
We are getting empty responses from the urls sometimes very randomly.
There is no firewall issue because we get the response most of the times.

**Comment by pcar75** [comment]
@Meemansa-Mishra  A warning message in red was displayed on the Explorer page ... on nov 10 and 11 saying that the London Linode server was having problems ; London AWS and an San Francisco were functioning fine ... Did you not see it ?!?!
The page layout had changed , having another dropbox ( left of 'Genre' dropbox ) for selecting which of the 3 directory server to use.

**Reply by rdica** [reply]
@pcar75 You can add custom directory server to the list in the web UI, this is what he's doing, and what I tested as well.

**Reply by Meemansa-Mishra** [reply]
@pcar75 
I did notice the new layout and tried the new urls, but getting empty response from those as well sometimes.

**Comment by ann0see** [comment]
I doubt the API displays the error message. I'd look at softins explorer repo. As far as I remember, there's a way to query non default backends. 

Edit: you wrote that you got errors with the new URLs. Missed that. 

I'd open an issue on the explorer repo as only softins can do something.

**Reply by Meemansa-Mishra** [reply]
@ann0see 
Thanks.
Please help me out if there is a another consistent way for me to get the list of servers.

**Reply by ann0see** [reply]
You could probably run the PHP script yourself on your server? It probably depends on your use case.

**Reply by Meemansa-Mishra** [reply]
I have tried doing this as well, but response from that is also not consistent.Facing the same issue there as well.

**Reply by ann0see** [reply]
Maybe the python scripts by "passing" can be used. You'll need to search his profile for them.

**Comment by mcfnord** [comment]
1. There are occasional, brief interruptions where the sampler produces an empty result for just one or two directories. I think these occasional interruptions have occurred ever since I arrived (2+ years ago).
2. A new problem causes the Linode UK sampler to return empty results for at least half directories, almost all of the time. Because (1) is true, I wonder if (2) is just a more extreme case of (1).
3. There are rare, brief outages that impact all samplers, and also impact Jamulus clients. Yesterday there was an interruption of this nature.
5. I can only guess that private directories face similar QoS failures.
6. It might be reasonable to see an empty result set as a failure, except maybe that's indistinguishable from an unpopulated directory.

https://jamulus.live places an Oops! line first when it notices an empty query on the SF sampler. It queries all the directories at once which might be useful in detecting larger outages.

I want to run servers.php in a debugger for a long while, to witness these scenarios.

**Reply by pljones** [reply]
What I'd like to see done is running queries against all the backends "concurrently" (as much as possible) and see whether the results differ.  The point being whether the "failures" are down to the Directory being queried or the routing of the request or response traffic.

I don't know how the PHP script works - the Jamulus Client, of course, "retries" Directory queries, so partial responses and timeouts are recoverable errors... so the above may or may not help.

**Comment by ann0see** [comment]
Pinging @softins he's the author of explorer.

**Comment by softins** [comment]
> I have used "**https://explorer.jamulus.io/servers.php?directory=**" url to get a list of all hosted servers on my directory but have been having a lot of trouble recently because I don't receive response from the url everytime.
> 
> I also noticed same problems with the new urls:

Please report details on my github page for the Explorer backend, https://github.com/softins/jamulus-php

I will need to know the exact invocations you use, including the IP or hostname of the directory, so I can try and reproduce the issue. If that information is sensitive and you want to keep it private, you can email it to me directly. Click on my name in the Explorer footer, and find my email address on my personal webpage.
